037-auto-rebalance-cluster-scaling.md

```diff
@@ -0,0 +1,265 @@
+# Automatic rebalance on cluster scaling
+
+This feature is about enabling the cluster operator to get an optimization proposal, which is auto-approved, in order to run the corresponding rebalancing when the user scales the cluster up or down by changing the `spec.kafka.replicas` field on the `Kafka` custom resource.
+Such a rebalancing can happen right after the scale up to move some replicas to the newly added brokers or right before the scale down to move replicas off of the brokers to be removed.
+
+## Current situation
+
+Currently, there is no automatic rebalancing when the user scales the cluster up or down.
+It means that:
+
+* after scaling up the cluster, in order to rebalance the cluster and move some replicas to the newly added brokers, the user has to create a `KafkaRebalance` custom resource using the `spec.mode: add-brokers` mode and add the new broker IDs to the the `spec.brokers` field.
```

Comment:
WDYT it should be a _partition_? You are not moving partitions. You moving replicas or partition-replicas if you want. Partition is a virtual thing which consists of one or more replicas. It is the replicas what is assigned to some broker and is moved.

Reply:
I know that CC uses replicas, which is a better term, but it is somewhat confusing in this context because we also have broker replicas. Maybe we could use partition replicas, as you suggests.

Reply:
Agree going to use "partition replicas" everywhere.